src/NetTopologySuite/Elevation/IElevationModel.cs

```diff
@@ -0,0 +1,54 @@
+using NetTopologySuite.Geometries;
+
+namespace NetTopologySuite.Algorithm.Elevation
+{
+    /// <summary>
+    /// A elevation model interface used to populate missing Z values.
+    /// </summary>
+    public interface IElevationModel {
```

Comment:
I can agree that `IElevationModel` is bloated which is because I aimed to lift out existing implementation with minimal changes. In hindsight that might be the wrong approach for a public interface.

However, making a completely new interface/abstraction will be a much larger effort than I had in mind. My original goal was to provide the ability to turn off the interpolation to make it possible to get the same behaviour as older NTS versions.

I'm unsure if I will find the time and motivation to go back to the drawing board and come up with a good general purpose public interface for general Z interpolation.

Reply:
One possible alternative is to go back to provide a plain and simple boolean option to opt out of interpolation instead of trying to find an optimal interface at this time. I would be willing to rework to that if that is agreed a better way forward.

Reply:
> My original goal was to provide the ability to turn off the interpolation to make it possible to get the same behaviour as older NTS versions (i.e no interpolation).
>
> One possible alternative is to go back to provide a plain and simple boolean option to opt out of interpolation instead of trying to find an optimal interface at this time. I would be willing to rework to that if that is agreed a better way forward.

I like the idea of adding a boolean flag to the offending operations to address the problem at hand, if part of the goal isn't **actually** to try to make something general-purpose — I had assumed that this was at least part of the intention, because of the general-sounding names and putting it prominently on `NtsGeometryServices`.

> I'm unsure if I will find the time and motivation to go back to the drawing board and come up with a good general purpose public interface for general Z interpolation.

I really don't want to stand in the way of progress — especially since I have had very little time + motivation to make any significant positive contributions of my own for a while now.

New public APIs are essentially permanent decisions, so it feels like **something** should change here... perhaps some names could be changed to make it clearer that types are intended for something more specific than what their current names suggest?  As long as someone won't mistake this for an attempt to make a general-purpose solution to this problem, I really can't justify being bothered by an imperfect API shape.

Reply:
Perfectly valid input @airbreather, and appreciated. Let's have it sink in a while and see what @FObermaier thinks about how/if to drive this forward.

Reply:
@airbreather is correct that the interface for elevation model is bloated with things that does not belong there. I have looked at it a bit more and think we can do better easily:
* Remove all but `GetZ` and `PopulateZ` from the interface. Add `Extent` property
* Create private `ZHandler` class in `RobustLineIntersector` that does all the things specific to `RobustLineIntersector` and have been removed from the `IElevationModel` interface. 
* Derive `ElevationModelZHandler` from it that uses the elevation model instead of interpolating z-ordinate values.

If you let me push to your repo @bjornharrtell we can stay in this PR

Reply:
@FObermaier feel free to commit to my repo/branch. I've invited you as collab.

Reply:
I just commited what I came up with. A few notes:
* It seems to me that `ElevationModel` should be a property of `GeometryFactory` and that `NtsGeometryServices` should provide means to automatically add one at construction time or get one for a given SRID and Extent/Geometry.
* Original `ElevationModel` should move back to `Operations.OverlayNG` namespace. That way it is easier to track changes.
* Should `BaseElevationModel` should be renamed to `ElevationModel`?
* Instead of having private `ZHandler` classes in RobustLineIntersector we could override RobustLineIntersector with a version that uses an elevation model to retrieve z-ordinate values. Might be cleaner and faster. This would involve having to replace every construction of `RobustLineIntersector` and replace it with some construction strategy based on the availability of an elevation model. But that would have to be done anyway eventually...